### ISSUE TITLE
LSM compact results are different among c api and java api

### ISSUE THREAD

**Issue by flyaway921** [wiredtiger/wiredtiger#1556]
Hi all ,
       I am reporting this issue because I think I have hit a bug. I modified the example code ex_thread.java and ex_thread.c with lsm mode to let it insert 1.2G data and then call compact api . The lsm files are compacted and merged into one lsm file when calling the c api. But the same case with java api doesn't get the same result --compact called and lsm files reduced but not merged into one lsm file. 
      The java api and c api makes different result, .I wonder if this is a bug of jni ,does anyone hit this?

Test env: 
Wiredtiger version: 2.5.0

Wei

**Comment by ddanderson** [comment]
Hello @flyaway921, can you post or send your example programs (or diffs) so we can see why they would be different?  Thanks!

**Comment by ddanderson** [comment]
@flyaway921, you say you modified the thread examples.  Are you doing insertions in parallel, or compaction while another thread inserts?  I'm just wondering if there may be some timing differences due to different threading models in the two languages.  Is it repeatable?  Thanks!

**Comment by ddanderson** [comment]
Closing until we hear back.
